docs/ai/how-to/swap-between-openai-and-aoai.md

```diff
@@ -0,0 +1,341 @@
+---
+title: "Swap between OpenAI and AOAI"
+description: "Demonstrates how to swap between OpenAI and Azure OpenAI (AOAI) using the Semantic Kernel SDK for .NET."
+author: haywoodsloan
+ms.topic: how-to
+ms.date: 04/09/2024
+
+#customer intent: As a .NET developer, I want to swap between OpenAI and Azure OpenAI (AOAI) so that I can choose the AI service that's best for me.
+
+---
+
+# Swap between OpenAI and AOAI
+
+This article demonstrates how to swap between the OpenAI and [Azure OpenAI](https://learn.microsoft.com/azure/ai-services/openai/overview) (AOAI) services using the [Semantic Kernel SDK](https://learn.microsoft.com/semantic-kernel/overview). Additionally, it shows how to switch your .NET projects to the Semantic Kernel SDK from the [AOAI client library](https://learn.microsoft.com/dotnet/api/overview/azure/ai.openai-readme).
+
+## Prerequisites
+
+* An Azure account with an active subscription. [Create an account for free](https://azure.microsoft.com/free/?WT.mc_id=A261C142F).
+* [.NET SDK](https://dotnet.microsoft.com/download/visual-studio-sdks)
+* [`Microsoft.SemanticKernel` Nuget package](https://www.nuget.org/packages/Microsoft.SemanticKernel)
+* [Create and deploy an Azure OpenAI Service resource](https://learn.microsoft.com/azure/ai-services/openai/how-to/create-resource)
+
+## Switch from the AOAI client library to Semantic Kernal SDK
+
+The following section shows how to add the Semantic Kernal SDK to your .NET project and use it to replace the AOAI client library.
+
+To begin add the [Semantic Kernal Nuget package](https://www.nuget.org/packages/Microsoft.SemanticKernel) to your project. The following command will add this package to the .NET project in your current working directory:
+
+```dotnetcli
+dotnet add package Microsoft.SemanticKernel
+```
+
+The Semantic Kernel SDK supports both prompt and chat completions:
+
+### Implement prompt completions with Semantic Kernel
+
+A basic implementation of prompt completions with the AOAI client library might look like the following:
+
+```csharp
+// === Retrieve the secrets saved during the Azure deployment ===
```

Comment:
Can you put code snippets into separate code files with a project file, so compilation is checked automatically as part of the PR?

Reply:
@alexwolfmsft is this something we want to do now or save for latter?